Azure.ResourceManager.Core/Resources/ResourceIdentifier.cs

```diff
@@ -21,10 +21,7 @@ public class ResourceIdentifier : IEquatable<ResourceIdentifier>, IEquatable<str
         public ResourceIdentifier(string id)
         {
             Id = id;
```

Comment:
let's add null check/verification at any public method entry:               if (string.IsNullOrWhiteSpace(id)) throw new ArgumentNullException(nameof(id));

Reply:
Null check will be added when we decided on #118